test/model/autocomplete_test.dart

```diff
@@ -0,0 +1,36 @@
+import 'package:checks/checks.dart';
+import 'package:test/scaffolding.dart';
+import 'package:zulip/api/model/model.dart';
+import 'package:zulip/model/autocomplete.dart';
+
+import '../example_data.dart' as eg;
+
+void main() {
+  test('MentionAutocompleteQuery.testUser', () {
+    doCheck(String rawQuery, User user, bool expected) {
+      final result = MentionAutocompleteQuery(rawQuery).testUser(user);
+      expected ? check(result).isTrue() : check(result).isFalse();
+    }
+
+    doCheck('', eg.user(fullName: 'Full Name'), true);
+    doCheck('Full Name', eg.user(fullName: 'Full Name'), true);
+    doCheck('full name', eg.user(fullName: 'Full Name'), true);
+    doCheck('Full Name', eg.user(fullName: 'full name'), true);
+    doCheck('Full', eg.user(fullName: 'Full Name'), true);
+    doCheck('Name', eg.user(fullName: 'Full Name'), true);
+    doCheck('Full Name', eg.user(fullName: 'Fully Named'), true);
+    doCheck('Full Four', eg.user(fullName: 'Full Name Four Words'), true);
```

Comment:
Also test that "F Full" and "Four F" don't match this name, while "Full F" and "F Four" do.